engine/transaction/record/record_outline_test.go

```diff
@@ -278,46 +259,78 @@ func TestRecordOutlineOpReturnErrorCases(t *testing.T) {
 			},
 			expectErr: txerrors.ErrAnnotationMismatch,
 		},
-		"error during broadcasting": {
-			broadcaster: newMockBroadcaster().withError(errors.New("broadcast error")),
-			repo: newMockRepository().withOutput(database.Output{
-				TxID: givenTXWithOpReturn(t).InputUTXO(0).TxID,
-				Vout: givenTXWithOpReturn(t).InputUTXO(0).Vout,
-			}),
-			outline: &outlines.Transaction{
-				BEEF: givenTXWithOpReturn(t).BEEF(),
-				Annotations: transaction.Annotations{
-					Outputs: transaction.OutputsAnnotations{
-						0: &transaction.OutputAnnotation{
-							Bucket: bucket.Data,
-						},
-					},
-				},
-			},
-			expectErr: txerrors.ErrTxBroadcast,
-		},
 	}
 	for name, test := range tests {
 		t.Run(name, func(t *testing.T) {
 			// given:
-			service := record.NewService(tester.Logger(t), test.repo, test.broadcaster)
-			initialOutputs := test.repo.getAllOutputs()
-			initialData := test.repo.getAllData()
+			given, then := testabilities.New(t)
+			service := given.WithStoredOutputs(test.storedOutputs...).NewRecordService()
 
 			// when:
 			err := service.RecordTransactionOutline(context.Background(), test.outline)
 
 			// then:
-			require.Error(t, err)
-			require.ErrorIs(t, err, test.expectErr)
-
-			// ensure that no changes were made to the repository
-			require.ElementsMatch(t, initialOutputs, test.repo.getAllOutputs())
-			require.ElementsMatch(t, initialData, test.repo.getAllData())
+			then.WithErrorIs(err, test.expectErr).NothingChanged()
 		})
 	}
 }
 
+func TestOnBroadcastErr(t *testing.T) {
+	// given:
+	given, then := testabilities.New(t)
+	service := given.
+		WithStoredOutputs(database.Output{
+			TxID:       givenTXWithOpReturn(t).InputUTXO(0).TxID,
+			Vout:       givenTXWithOpReturn(t).InputUTXO(0).Vout,
+			SpendingTX: nil,
+		}).
+		WillFailOnBroadcast(errors.New("broadcast error")).
+		NewRecordService()
```

Comment:
I would rather add context to calls, to increase understandability what is the environment setup

```suggestion
	given.Repository().HasOutputs(database.Output{
			TxID:       givenTXWithOpReturn(t).InputUTXO(0).TxID,
			Vout:       givenTXWithOpReturn(t).InputUTXO(0).Vout,
			SpendingTX: nil,
		})
	given.Broadcaster().WillFailOnBroadcast(errors.New("broadcast error")).
		
	service := given.NewRecordService()
```



or:

```suggestion
	given.Repository().HasOutputs(database.Output{
			TxID:       givenTXWithOpReturn(t).InputUTXO(0).TxID,
			Vout:       givenTXWithOpReturn(t).InputUTXO(0).Vout,
			SpendingTX: nil,
		}).
	And().Broadcaster().WillFailOnBroadcast(errors.New("broadcast error")).
		
	service := given.NewRecordService()
```

Reply:
I added `Repository` and `Broadcaster` methods - so the `given` test configuration is setup without `RecordServiceFixture` as "proxy"